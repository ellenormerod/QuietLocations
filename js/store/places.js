import axios from 'axios';

//Action types
const GET_PLACES = 'GET_PLACES';

//Action creators
const getPlaces = places => ({type: GET_PLACES, places})

//Thunk creators
export const fetchPlaces = () => dispatch => {
  return axios.get('http://localhost:8080/api/places')
  .then(res => {
    return dispatch(getPlaces(res.data))})
  .catch(err => console.error(err))
  
}

//Reducer
export default function(places = [], action){
  switch (action.type){
    case GET_PLACES:
      return action.places
    default:
      return places
  }
}