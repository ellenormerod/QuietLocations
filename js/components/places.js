import React, { Component } from 'react';
import {
  Text,
  View,
  FlatList,
  StyleSheet
} from 'react-native';
import { List, ListItem, SearchBar } from 'react-native-elements';
import { fetchPlaces } from '../store';
import { connect } from 'react-redux';
import Place from './place';
import { StackNavigator } from 'react-navigation';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});

class Places extends Component {
  componentDidMount() {
    this.props.loadInitialData()
  }

  renderSeparator = () => {
    return (
      <View
        style={{
          height: 1,
          width: '86%',
          backgroundColor: '#CED0CE'
        }}
      />
    )
  }

  renderHeader = () => {
    return <SearchBar placeholder="Search here" lightTheme round />
  }


  render() {
    const { places } = this.props
    return (
      <List containerStyle={{ borderTopWidth: 0, borderBottomWidth: 0 }}>
        <FlatList
          data={places}
          renderItem={({ item }) => (
            <ListItem
              title={item.name}
              subtitle={item.location}
              containerStyle={{ borderBottomWidth: 0 }}
            />
          )}
          keyExtractor={item => item.name}
          ItemSeparatorComponent={this.renderSeparator}
          ListHeaderComponent={this.renderHeader}
        />
      </List>
    )
  }
}

const mapState = state => {
  return {
    places: state.places
  }
}

const mapDispatch = dispatch => {
  return {
    loadInitialData() {
      dispatch(fetchPlaces())
    }
  }
}


export default connect(mapState, mapDispatch)(Places)
