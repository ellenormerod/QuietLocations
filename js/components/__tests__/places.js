import React from 'react';
import { Places } from '../places';
import renderer from 'react-test-renderer';

const fakeProps = {
  places: {
    name: 'McDonalds',
    location: '145 W Cool Street St. Chicago, IL',
    noise: 3
  },
  loadInitialData: () => { }
}


it('renders Places using Snapshot', () => {
  expect(renderer.create(
    <Places {...fakeProps} />
  )).toMatchSnapshot()
})
