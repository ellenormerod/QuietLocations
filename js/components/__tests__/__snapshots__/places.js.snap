// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Places using Snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      Object {
        "borderBottomWidth": 0,
        "borderTopWidth": 0,
      },
    ]
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListHeaderComponent={[Function]}
    data={
      Object {
        "location": "145 W Cool Street St. Chicago, IL",
        "name": "McDonalds",
        "noise": 3,
      }
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#393e42",
                "borderBottomColor": "#000",
                "borderBottomWidth": 1,
                "borderTopColor": "#000",
                "borderTopWidth": 1,
              },
              Object {
                "backgroundColor": "#e1e8ee",
                "borderBottomColor": "#e1e1e1",
                "borderTopColor": "#e1e1e1",
              },
              undefined,
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            placeholder="Search here"
            placeholderTextColor="#86939e"
            style={
              Array [
                Object {
                  "backgroundColor": "#303337",
                  "borderRadius": 3,
                  "color": "#86939e",
                  "fontSize": 17.5,
                  "height": 30,
                  "margin": 8,
                  "overflow": "hidden",
                  "paddingLeft": 26,
                  "paddingRight": 19,
                },
                Object {
                  "backgroundColor": "#bdc6cf",
                },
                false,
                Object {
                  "borderRadius": 15,
                },
                undefined,
                undefined,
                false,
              ]
            }
            underlineColorAndroid="transparent"
          />
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#86939e",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "position": "absolute",
                    "top": 15.5,
                  },
                  Object {
                    "left": 16,
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            î¢¶
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
